code/modules/client/preferences/middleware/quirks.dm

```diff
@@ -31,6 +31,8 @@
 
 	var/list/quirks = SSquirks.get_quirks()
 
+	var/datum/species/user_species = preferences.read_preference(/datum/preference/choiced/species)
+
```

Comment:
```suggestion
```
This is the wrong time to look this up, plus it is not necessary to re-send this to the client.